rmgpy/molecule/molecule.py

```diff
@@ -1595,4 +1595,38 @@ def toGroup(self):
         
         return group
 
-
+    def getNthNeighbor(self, startingAtoms, n, blacklist):
+        '''
+        Recursively get the Nth nonHydrogen neighbors of the startingAtoms.
+        StartingAtoms should be a list.
+        Return a list.
+        '''
+        neighbors = []
+        for atom in startingAtoms:
+            for neighbor in self.getBonds(atom):
```

Comment:
Nit-nit-pick: the for loop is still there, just implicit, and thus (arguably) harder to understand 😉. Not sure if speed is an issue here, nor how these alternatives would compare, with or without cython. (Probably not worth worrying about)